ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -213,58 +221,120 @@ private void registerDefaultCommands() {
+    }
+
+    public static class Builder {
+      private final List<String> columnHeaders = new LinkedList<>();
+      private final List<List<String>> rowValues = new LinkedList<>();
+      private final List<String> header = new LinkedList<>();
+      private final List<String> footer = new LinkedList<>();
+
+      Builder() {}
+
+      public Table build() {
+        return new Table(columnHeaders, rowValues, header, footer);
+      }
 
-    header.forEach(m -> writer().println(m));
+      public Builder withColumnHeaders(List<String> columnHeaders) {
+        this.columnHeaders.addAll(columnHeaders);
+        return this;
+      }
 
-    if (columnHeaders.size() > 0) {
-      addResult(columnHeaders, rowValues);
+      public Builder withColumnHeaders(String... columnHeaders) {
+        this.columnHeaders.addAll(Arrays.asList(columnHeaders));
+        return this;
+      }
 
-      Integer[] columnLengths = new Integer[columnHeaders.size()];
-      int separatorLength = -1;
+      public Builder withRows(List<List<String>> rowValues) {
+        this.rowValues.addAll(rowValues);
+        return this;
+      }
 
-      for (int i = 0; i < columnLengths.length; i++) {
-        int columnLength = getColumnLength(columnHeaders, rowValues, i);
-        columnLengths[i] = columnLength;
-        separatorLength += columnLength + 3;
+      public Builder withRow(String... row) {
+        this.rowValues.add(Arrays.asList(row));
+        return this;
       }
 
-      String rowFormatString = constructRowFormatString(columnLengths);
+      public Builder withRow(List<String> row) {
+        this.rowValues.add(row);
+        return this;
+      }
 
-      writer().printf(rowFormatString, columnHeaders.toArray());
+      public Builder withHeaderLine(String headerLine) {
+        this.header.add(headerLine);
+        return this;
+      }
+
+      public Builder withFooterLine(String footerLine) {
+        this.footer.add(footerLine);
+        return this;
+      }
+    }
+
+    private int getMultiLineStringLength(String s) {
+      String[] split = s.split("\n");
+      String longest = Collections.max(
+          Arrays.asList(split),
+          Comparator.comparing(line -> line.length())
+      );
+      return longest.length();
+    }
 
-      writer().println(new String(new char[separatorLength]).replaceAll(".", "-"));
+    private int getColumnLength(List<String> columnHeaders, List<List<String>> rowValues, int i) {
+      int columnLength = columnHeaders.get(i).length();
       for (List<String> row : rowValues) {
-        writer().printf(rowFormatString, row.toArray());
+        columnLength = Math.max(columnLength, getMultiLineStringLength(row.get(i)));
       }
-      writer().println(new String(new char[separatorLength]).replaceAll(".", "-"));
+      return columnLength;
     }
 
-    footer.forEach(m -> writer().println(m));
+    public void print(final Console console) {
```

Comment:
Do we need to pass in `Console` here? Looks like we only need a `Writer`

Reply:
Good call. Originally there were some other functions in Console that this code was using that I moved into Table instead.